Dockerfile

```diff
@@ -1,6 +1,14 @@
 FROM ruby:2.3.3
 
-RUN apt-get update && apt-get install -y nodejs --no-install-recommends && rm -rf /var/lib/apt/lists/*
+RUN apt-get update && apt-get install apt-transport-https
+RUN mkdir /usr/local/node \
+    && curl -L https://nodejs.org/dist/v8.9.4/node-v8.9.4-linux-x64.tar.xz | tar Jx -C /usr/local/node --strip-components=1
+RUN ln -s ../node/bin/node /usr/local/bin/
+
+ADD https://dl.yarnpkg.com/debian/pubkey.gpg /tmp/yarn-pubkey.gpg
+RUN apt-key add /tmp/yarn-pubkey.gpg && rm /tmp/yarn-pubkey.gpg
+RUN echo 'deb https://dl.yarnpkg.com/debian/ stable main' > /etc/apt/sources.list.d/yarn.list
+RUN apt-get update && apt-get install -y --no-install-recommends yarn
```

Comment:
It seems like this probably results in a `node` package of some sort being installed, since it doesn't know about the non-`dpkg` copy of `node` installed above.

Reply:
I'd recommend installing Node like so: https://github.com/nodesource/distributions#debmanual

Reply:
Thanks. I will try that.

Reply:
That didn't work, but this did:
```
RUN curl -sL https://deb.nodesource.com/setup_8.x | bash -
RUN apt-get update && apt-get install -y nodejs
```